consensus/state.go

```diff
@@ -2248,7 +2266,10 @@ func (cs *State) updatePrivValidatorPubKey() error {
 		return nil
 	}
 
-	pubKey, err := cs.privValidator.GetPubKey()
+	// set a hard timeout for 2 seconds. This helps in avoiding blocking of the remote signer connection
+	ctx, cancel := context.WithTimeout(context.TODO(), 2*time.Second)
```

Comment:
Are we sure 2 seconds is sufficient. Do we have any heuristics for this?

Reply:
We dont have any data, I can test different values but with gRPC 2 seconds would be more than enough.

Reply:
In testing this works fine. Not sure how to get results to build more confidence. I can leave a comment mentioning this is double the default time of prevote/precommit timeouts.